ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1284,56 +1292,60 @@ RouterErrorCode processServerError(ServerErrorCode errorCode) {
      */
     private void handleMetadataBlob(BlobData blobData, byte[] userMetadata, ByteBuffer encryptionKey)
         throws IOException, MessageFormatException {
-      ByteBuffer serializedMetadataContent = blobData.getStream().getByteBuffer();
-      compositeBlobInfo =
-          MetadataContentSerDe.deserializeMetadataContentRecord(serializedMetadataContent, blobIdFactory);
-      totalSize = compositeBlobInfo.getTotalSize();
-      chunkMetadataList = compositeBlobInfo.getChunkMetadataList();
-      boolean rangeResolutionFailure = false;
+      ByteBuf serializedMetadataContent = blobData.getAndRelease();
       try {
-        if (options.getBlobOptions.hasBlobSegmentIdx()) {
-          int requestedSegment = options.getBlobOptions.getBlobSegmentIdx();
-          if (requestedSegment < 0 || requestedSegment >= chunkMetadataList.size()) {
-            throw new IllegalArgumentException(
-                "Bad segment number: " + requestedSegment + ", num of keys: " + chunkMetadataList.size());
+        compositeBlobInfo =
+            MetadataContentSerDe.deserializeMetadataContentRecord(serializedMetadataContent.nioBuffer(), blobIdFactory);
+        totalSize = compositeBlobInfo.getTotalSize();
+        chunkMetadataList = compositeBlobInfo.getChunkMetadataList();
+        boolean rangeResolutionFailure = false;
```

Comment:
Are the only changes after this point in this block indentation from the added try-finally block?

Reply:
that's right. It's not perfectly lined up.